resources/s3-object-versions.go

```diff
@@ -0,0 +1,33 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3ObjectVersion struct {
```

Comment:
I wouldn't create a new type for that. Currently the `S3Nuke.ListObjects` produces two types which are `S3ObjectVersion` and `S3Object`.

The problem there is that the filtering and refreshing currently relies on weird reflections of the type. This would work for now, but I want to get rid of this reflection code and make things more explicit (#66).

This means that one type should return exactly one type. Otherwise filtering and refreshing would get complicated, because *aws-nuke* would need to scan all resources on every iteration. This would slow down the execution and cause an unnecessary usage of the AWS API.

So my suggestion would be to merge these two types and make the `versionID` optional (by converting it to a reference). This is basically what the AWS SDK already does.

Reply:
I am not too sure, to do this right can be quite tricky and I could image a couple of edge cases. That's why I decided to use a separate type for it.

Happy to give the integrated type a try, but won't guarantee for anything  :laughing: